From bf157f5b6bbb245f41a805b438f96511e19ed5de Mon Sep 17 00:00:00 2001
From: template <nomail@local>
Date: Fri, 6 Sep 2019 08:29:50 +0000
Subject: [PATCH] hostapd: disable 802.11 legacy-rates by default

In https://github.com/lede-project/source/commit/e194e1b3c838a301178effb639804c28fe67354d
the option "legacy-rates" was added to uci.wireless.radio. So on
a per radio-basis the 802.11b rates can be disabled easily.
LEDE implemented this as having legacy-rates enabled by default, but
we want to have them disabled by default. (see https://github.com/freifunk-berlin/firmware/issues/375)

So we change the default of LEDE. This also safes some work in
changing the UCI-config on every node.
The node-operator is always free to enable these rates again by
adding the legacy_rates option.

This was also tested on a 5GHz radio of a Dualband-router and
did no show any side-effects in this band.
---
 package/network/services/hostapd/files/hostapd.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/package/network/services/hostapd/files/hostapd.sh b/package/network/services/hostapd/files/hostapd.sh
index 189760157e..df6162efe9 100644
--- a/package/network/services/hostapd/files/hostapd.sh
+++ b/package/network/services/hostapd/files/hostapd.sh
@@ -116,7 +116,7 @@ hostapd_prepare_device_config() {
 	set_default country_ie 1
 	set_default spectrum_mgmt_required 0
 	set_default doth 1
-	set_default legacy_rates 1
+	set_default legacy_rates 0
 
 	[ "$hwmode" = "b" ] && legacy_rates=1
 
-- 
2.20.1

